153_save_sensor_data/sensor_data_store.py

```diff
@@ -0,0 +1,238 @@
+            interpolated_set = interpolate_linear(
+                new_time,
+                # Time and values of the largest pair
+                largest_set_without_going_over,
+                # Time and values of the smallest pair
+                smallest_set_without_going_under
+            )
+            interpolated_data.append(interpolated_set)
+            if debugInterpolatedData: # print new data to screen to debug sensor
+                print(f"Time: {interpolated_set['seconds']:<10.2f} "
+                      f"CO2: {interpolated_set['co2']:>6.1f} ppm    "
+                      f"T: {interpolated_set['temp']:<3.2f}°C    "
+                      f"Humidity: {interpolated_set['humidity']:<3.2f}%")
+            interpolated_time += desired_time_step
+        
+        # If more than WRITE_QTY data points have been gathered, write to file
+        if len(interpolated_data) > WRITE_QTY:
+
+            column_names = ['time', 'co2','temp','humidity']
+            file_name = f"scd_data/raw/raw_data{interpolated_time}.csv"
+            #Write Raw Data
+            with open(file_name, 'w', newline='') as csvfile:
+                csv_writer = csv.DictWriter(csvfile, sensor_data[0].keys())
+                csv_writer.writeheader()
+                csv_writer.writerows(sensor_data)
+            # Write Interpolated Data
+            file_name = f"scd_data/interpolated/interpolated_data{interpolated_time}.csv"
+            with open(file_name, 'w', newline='') as csvfile:
+                csv_writer = csv.DictWriter(csvfile, interpolated_data[0].keys())
+                csv_writer.writeheader()
+                csv_writer.writerows(interpolated_data)
+
+            # Reset raw and interpolated data
+            # Remove all the data points smaller than the interpolated time
+            # Leaving the extra for the next interpolation.
+            total_vals = len(sensor_data)
+            i = total_vals - 1
+            while i >= 0 :
+                if sensor_data[i]['seconds'] < interpolated_time:
+                    sensor_data.remove(sensor_data[i])
+                i -= 1
+            # Just like in Java, this doesn't work in Python:
+            # for i, data_set in enumerate(sensor_data): # Keep some data for interpolation
+            #    if ( data_set['seconds'] < interpolated_data[-2]['seconds']):
+            #        sensor_data.remove(data_set) # Remove old data
+            interpolated_data.clear()
+
+        # New sensor data is only available once every 2 seconds by default,
+        # but here we poll every 0.5 seconds by default, 
+        # (or 4 times the measurement interval), to try and catch every
+        # available update, even if there was a failure to reach the SCD-30.
+        time.sleep(scd.measurement_interval/4)
+       
+
+#Start the sensor
+os.system("mkdir scd_data")
+os.system("mkdir scd_data/raw")
+os.system("mkdir scd_data/interpolated")
```

Comment:
You shouldn't use `os.system()` in most cases.  If you want to execute shell commands, you can use `subprocess.run()` but for common operations like creating dirs there are already Python functions that do that in a portable way.  In particular look at the `pathlib` (you should use this for any other path manipulation too) and `shutil` modules.

Reply:
pathlib has now been implemented.